src/Listener/WebhookListener.php

```diff
@@ -0,0 +1,166 @@
+    public function __construct(ServiceLocatorInterface $serviceLocator)
+    {
+        $this->serviceLocator = $serviceLocator;
+        $this->moduleOptions  = $serviceLocator->get(ModuleOptions::class);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function attach(EventManagerInterface $eventManager)
+    {
+        $sharedManager = $eventManager->getSharedManager();
+        $sharedManager->attach(WebhookListenerController::class, WebhookEvent::WEBHOOK_RECEIVED, [$this, 'dispatchWebhook']);
+    }
+
+    /**
+     * @internal
+     * @param  WebhookEvent $event
+     * @return string
+     */
+    public function dispatchWebhook(WebhookEvent $event)
+    {
+        switch ($event->getEventName()) {
+            case 'customer.discount.created':
+            case 'customer.discount.updated':
+            case 'customer.discount.deleted':
+                return $this->handleDiscountEvent($event->getEvent());
+
+            case 'customer.subscription.updated':
+            case 'customer.subscription.deleted':
+                return $this->handleSubscriptionEvent($event->getEvent());
+
+            case 'plan.created':
+            case 'plan.updated':
+            case 'plan.deleted':
+                return $this->handlePlanEvent($event->getEvent());
+
+            case 'invoice.created':
+            case 'invoice.updated':
+            case 'invoice.payment_succeeded':
+            case 'invoice.payment_failed':
+                return $this->handleInvoiceEvent($event->getEvent());
+
+            default:
+                return ''; // Any other event is not handled by default
+        }
+    }
+
+    /**
+     * Handle a discount Stripe event
+     *
+     * @param  array $stripeEvent
+     * @return string
+     */
+    public function handleDiscountEvent(array $stripeEvent)
+    {
+        /** @var DiscountService $discountService */
+        $discountService = $this->serviceLocator->get(DiscountService::class);
+        $discountService->syncFromStripeEvent($stripeEvent);
+
+        return 'Event has been properly processed';
+    }
+
+    /**
+     * Handle a subscription Stripe event
+     *
+     * @param  array $stripeEvent
+     * @return string
+     */
+    public function handleSubscriptionEvent(array $stripeEvent)
+    {
+        /** @var SubscriptionService $subscriptionService */
+        $subscriptionService = $this->serviceLocator->get(SubscriptionService::class);
+        $subscriptionService->syncFromStripeEvent($stripeEvent);
+
+        return 'Event has been properly processed';
+    }
+
+    /**
+     * Handle a plan Stripe event
+     *
+     * @param  array $stripeEvent
+     * @return string
+     */
+    public function handlePlanEvent(array $stripeEvent)
+    {
+        /** @var PlanService $planService */
+        $planService = $this->serviceLocator->get(PlanService::class);
+        $planService->syncFromStripeEvent($stripeEvent);
+
+        return 'Event has been properly processed';
```

Comment:
This is human-readable text: doesn't seem to be the correct response for an event handler IMO